modules/account-tokenizer/src/lib.rs

```diff
@@ -0,0 +1,498 @@
+	impl<T: Config> GenesisBuild<T> for GenesisConfig {
+		fn build(&self) {
+			Pallet::<T>::on_runtime_upgrade();
+		}
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		/// Create the NFT class once.
+		fn on_runtime_upgrade() -> Weight {
+			let on_chain_storage_version = <Self as GetStorageVersion>::on_chain_storage_version();
+			if on_chain_storage_version == 0 {
+				let create_class_cost = T::NFTInterface::base_create_class_fee();
+
+				// Transfer some fund from the treasury to pay for the class creation.
+				let res = T::Currency::transfer(
+					&T::TreasuryAccount::get(),
+					&Self::account_id(),
+					create_class_cost,
+					KeepAlive,
+				);
+				log::debug!(
+					"Account Tokenizer: Transferred funds from treasury to create class. result: {:?}",
+					res
+				);
+
+				// Use storage version to ensure we only register NFT class once.
+				let class_id = T::NFTInterface::next_class_id();
+				let res = T::NFTInterface::create_class(&class_id, &Self::account_id(), &Self::account_id());
+				log::debug!("Account Tokenizer: Created NFT class. result: {:?}", res);
+
+				let res = T::NFTInterface::set_class_properties(
+					&class_id,
+					Properties(
+						ClassProperty::Transferable
+							| ClassProperty::Burnable | ClassProperty::Mintable
+							| ClassProperty::ClassPropertiesMutable,
+					),
+				);
+				log::debug!("Account Tokenizer: Set NFT class property. result: {:?}", res);
+
+				// Sets NFT class ID storage
+				NFTClassId::<T>::put(class_id);
+
+				// Upgrade storage versino so NFT class is only created once.
+				StorageVersion::new(1).put::<Self>();
+				<T as Config>::WeightInfo::initialize_nft_class()
+			} else {
+				0
+			}
+		}
+
+		// ensure that MintFee is >= NFT's mint fee.
+		fn integrity_test() {
+			sp_std::if_std! {
+				sp_io::TestExternalities::new_empty().execute_with(||
+					assert!(
+						T::MintFee::get() >= T::NFTInterface::base_mint_fee()
+					));
+			}
+		}
+	}
+
+	#[pallet::pallet]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Request to mint an Account Token. Called after the user of the same Account Id has given
+		/// the proxy control of an account to the parachain account.
+		///
+		/// Params:
+		/// 	- `account`: The account ID of the anonymous proxy.
+		/// 	- `original_owner`: The original owner's account ID. Used to verify anonymous proxy.
+		/// 	- `height`: The block number in which the anonymous proxy is generated.
+		/// 	- `ext_index`: The index, in the block, of the extrinsics that generated the anonymous
+		///    proxy.
+		/// 	- `index`: The index of the anonymous proxy.
+		#[pallet::weight(< T as Config >::WeightInfo::request_mint())]
+		#[transactional]
+		pub fn request_mint(
+			origin: OriginFor<T>,
+			account: T::AccountId,
+			original_owner: T::AccountId,
+			height: T::BlockNumber,
+			ext_index: u32,
+			index: u16,
+		) -> DispatchResult {
```

Comment:
Right now anyone can `request_mint`. There needs to be a solution to not make it a race to who can mint first. My original solution was to only let the account that spawned the anon proxy call `request_mint`. I'm open to other solutions that may be a bit better to allow minting of accounts a user didn't spawn, but need some sort of auth system so only the owner of anon proxy can mint nft

Reply:
what should be happening is that bad request should be slashed. there should only be one owner of the NFT and if the owner is making multiple requests, then one of it will success and rest should be slashed.

Reply:
On the relay chain how do we know who owns the Anon Proxy?

Reply:
I was about to say use the last `ProxyRemoved` event as owner and only request from this account is valid. But there is no such event yet... Let me make a PR to Substrate.
We definitely need a way to only allow the last owner of the anon proxy to be able to mint.

Reply:
https://github.com/paritytech/substrate/pull/11085

Reply:
Still isn't ideal because then we need oracle to read historical state

Reply:
Right. The other way is the owner don't remove itself from the proxy. So the anon proxy will have two owner, the parachain, and the original owner. And then we can verify that's the case and send XCM to ensure the parachain is the solo owner